src/main/kotlin/com/msd/command/application/CommandApplicationService.kt

```diff
@@ -0,0 +1,169 @@
+     *
+     * @throws CommandParsingException if there are not enough arguments for the command type or if the command type
+     *         is unknown
+     * @param commandString The string representation of a command
+     * @return The corresponding Command object
+     */
+    fun parseCommand(commandString: String): Command {
+        val (verb, args) = getVerbAndArgs(commandString)
+
+        if (verb !in commandTypeKeywordsAndNumArgs.keys) throw CommandParsingException(commandString)
+
+        if (args.count() != commandTypeKeywordsAndNumArgs[verb]) throw CommandParsingException(commandString)
+
+        try {
+            return if (verb.startsWith("use-item-"))
+                parseItemUsageCommand(verb, args)
+            else
+                parseActionCommand(verb, args)
+        } catch (iae: IllegalArgumentException) {
+            if (iae.message?.contains("Invalid UUID") == true)
+                throw CommandParsingException(commandString, "Invalid UUID string")
+            else throw CommandParsingException(commandString, "Unknown ItemType: ${args[2]}")
+        }
+    }
+
+    /**
+     * Parse the command string into the verb and the remaining arguments.
+     *
+     * @param commandString: The string containing a command
+     * @return Pair of verb and args
+     */
+    private fun getVerbAndArgs(commandString: String): Pair<String, List<String>> {
+        val parts = commandString
+            .lowercase()
+            .replace("\\s+".toRegex(), " ")
+            .split(" ")
+            .filter { it.isNotBlank() }
+        val verb = parts.first()
+        val args = parts.takeLast(parts.size - 1)
+        return Pair(verb, args)
+    }
+
+    /**
+     * Parse all commands describing the usage of an item
+     *
+     * @param verb  The command verb, describing the type of command
+     * @param args  The arguments for the command
+     */
+    private fun parseItemUsageCommand(verb: String, args: List<String>): Command {
+        when (verb) {
+            "use-item-fighting" -> return AttackItemUsageCommand(
+                UUID.fromString(args[0]),
+                UUID.fromString(args[1]),
+                AttackItemType.valueOf(args[2].uppercase()),
+                UUID.fromString(args[3]),
+                UUID.fromString(args[4])
+            )
+            "use-item-movement" -> return MovementItemsUsageCommand(
+                UUID.fromString(args[0]),
+                UUID.fromString(args[1]),
+                MovementItemType.valueOf(args[2].uppercase()),
+                UUID.fromString(args[3])
+            )
+            "use-item-healing" -> return ReparationItemUsageCommand(
+                UUID.fromString(args[0]),
+                UUID.fromString(args[1]),
+                ReparationItemType.valueOf(args[2].uppercase()),
+                UUID.fromString(args[3])
+            )
+        }
+        throw RuntimeException("Internal Error")
```

Comment:
Why is this exception here and when is it thrown? There also should be a default case in the `when` expression. The same applies to all other `when` expressions in this file.